vim-plugin/lua/changetracker.lua

```diff
@@ -2,15 +2,15 @@ local sync = require("vim.lsp.sync")
 local utils = require("utils")
 local debug = require("logging").debug
 
-local M = {}
+local changetracker = {}
```

Comment:
I have seen a couple of plugins which use the module name as their "module table" and I wanted to try it. Happy to revert these changes again.

The upside is that it reads nicely in the function definitions, the downside is that it makes the code a bit less mobile if we want to restructure some functions.

I guess it comes down to personal taste and just having a consistent style. Also I think it somewhat can prepare towards a more object oriented approach, not just having a "capital M" everywhere, but by thinking more explicitly "this is a changetracker function".

Reply:
I think I prefer the `M` approach, similar to how I prefer using `Self` in Rust whereever possible. Mostly, I think this makes renaming things easier/results in a smaller diff. But yes, it's a taste thing :)